WIP - Migrate listener package (#7)
kyma-project/kyma-watcher: docs/Listener.md

# Kyma Listener

> Listener component that listens to events sent by the Kyma [watcher](https://github.com/kyma-project/kyma-watcher) component.

Reviewer: are you using the words "component" and "module" as synonyms here, or is "listener component" (line 4) something different from "listener module" (line 8)? 

```suggestion
Listener component that listens to events sent by the Kyma [Watcher](https://github.com/kyma-project/kyma-watcher) component.
```
In the [Watcher README](https://github.com/kyma-project/kyma-watcher/tree/e84b0a21696f5b13e50451990bfb332cfe0f9ea7#readme), you capitalize the "Watcher". We should stay consistent here.
In a related note, should the "listener" be capitalized as well, or is there a reason to keep it lower case? Or shall we use lower case for both, "watcher" and "listener"?

Author: Thank you for pointing this out! Both should be capitalized, and the listener is a module, not a separate component.
I have changed all **_listener component_** occurrences to **_Listener module_**.

## Overview

> The listener module is typically used with operators built using kube-builder but its use is not resticted only to that.

Author: Thank you for the changing the phrasing! Your guess is perfect!
I have applied all of the suggestions 👍

### Use

> 

Reviewer: Are the first two steps optional, or why do they contain "might"?
Is the only mandatory step the third step?

Author: All the steps are mandatory. Should I remove “might” from the first two?

Reviewer: I've added two suggestions! :)

Author: Thank you! Applied them 👍

> 1. For operators built using the kube-builder framework, you might leverage your `SetupWithManager()` method to initialize the listener by calling `RegisterListenerComponent()`.

Reviewer: ```suggestion
1. For operators built using the Kubebuilder framework, you might leverage your `SetupWithManager()` method to initialize the listener by calling `RegisterListenerComponent()`.
```

2. You might also setup your controller to watch for changes sent through the `source.Channel{}` returned by the listener component and react to them calling the `(blder *Builder) Watches()` method and providing your `handler.EventHandler` implementation.

> 3. The last step is to start the listener, which is accomplished by adding the listener as a runnable to your controller-manager. This is done by calling `mgr.Add()` and passing the listener returned by `RegisterListenerComponent()`.

Reviewer: ```suggestion
3. To start the listener, add it as a runnable to your controller-manager: Call `mgr.Add()` and pass the listener returned by `RegisterListenerComponent()`.
```

# Kyma Listener

> Listener module that listens to events sent by the Kyma [Watcher](https://github.com/kyma-project/kyma-watcher) component.

Author: Thanks! Applied it 👍

## Overview

Typically, we use the Listener module with operators built using Kubebuilder, but we can use it for other operators as well.

### Use

> 1. For operators built using the Kubebuilder framework, you might leverage your `SetupWithManager()` method to initialize the Listener by calling `RegisterListenerComponent()`.

Reviewer: Not optional, dropping the "might" 

```suggestion
1. For operators built using the Kubebuilder framework, leverage your `SetupWithManager()` method to initialize the Listener by calling `RegisterListenerComponent()`.
```

> 2. You might also setup your controller to watch for changes sent through the `source.Channel{}` returned by the Listener module, and to react to them by calling the `(blder *Builder) Watches()` method and providing your `handler.EventHandler` implementation.

Reviewer: ```suggestion
2. Set up your controller to watch for changes sent through the `source.Channel{}` returned by the Listener module, and to react to them by calling the `(blder *Builder) Watches()` method and providing your `handler.EventHandler` implementation.
```
